services/s3/src/test/java/software/amazon/awssdk/services/s3/internal/crossregion/S3CrossRegionAsyncClientTest.java

```diff
@@ -437,18 +438,18 @@ void given_AnyRegion_Client_Updates_the_useGlobalEndpointFlag_asFalse(String reg
             .thenReturn(CompletableFuture.completedFuture(Endpoint.builder().url(URI.create("https://bucket.s3.amazonaws.com")).build()));
 
         S3AsyncClient s3Client = clientBuilder().crossRegionAccessEnabled(true)
-                                           .region(Region.of(region))
-                                           .endpointProvider(mockEndpointProvider).build();
+                                                .region(Region.of(region))
+                                                .endpointProvider(mockEndpointProvider).build();
         s3Client.getObject(r -> r.bucket(BUCKET).key(KEY), AsyncResponseTransformer.toBytes()).join();
         assertThat(captureInterceptor.endpointProvider).isInstanceOf(BucketEndpointProvider.class);
         ArgumentCaptor<S3EndpointParams> collectionCaptor = ArgumentCaptor.forClass(S3EndpointParams.class);
-        verify(mockEndpointProvider).resolveEndpoint(collectionCaptor.capture());
-        S3EndpointParams resolvedParams = collectionCaptor.getAllValues().get(0);
-        assertThat(resolvedParams.region()).isEqualTo(Region.of(region));
-        assertThat(resolvedParams.useGlobalEndpoint()).isFalse();
+        verify(mockEndpointProvider, atLeastOnce()).resolveEndpoint(collectionCaptor.capture());
```

Comment:
nit: should we just verify twice, non blocking, can be a follow up PR.

Reply:
Will push this once now to get in release , updating might cause checkbuild to start and cause delay , will create separate PR for times(2)